sys/net/routing/nhdp/nhdp.c

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     nhdp
+ * @{
+ *
+ * @file
+ * @brief       Implementation of NHDP's core functionality
+ *
+ * @author      Fabian Nack <nack@inf.fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "msg.h"
+#include "netapi.h"
+#include "thread.h"
+#include "utlist.h"
+
+#include "rfc5444/rfc5444_writer.h"
+
+#include "lib_table.h"
+#include "iib_table.h"
+#include "nib_table.h"
+#include "nhdp.h"
+#include "nhdp_address.h"
+#include "nhdp_writer.h"
+#include "nhdp_reader.h"
+
+char nhdp_stack[NHDP_STACK_SIZE];
```

Comment:
But you need to have a size defined before. `char ndhp_stack[sizeof(ndhp_stack)]` is either black magic or none-sense, since you would create a recursive paradox ;-).